src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -996,83 +1007,63 @@ public void updateServicePlan(String serviceName, String planName) {
+                                                                                               .build())
+                                                    .build())
                 .block();
     }
 
+    private CloudServiceInstance getServiceInstanceWithoutAuxiliaryContent(String serviceInstanceName) {
+        var serviceInstance = fetch(() -> getServiceInstanceResourceByName(serviceInstanceName), ImmutableRawCloudServiceInstance::of);
+        return getServiceInstanceIfRequired(serviceInstanceName, serviceInstance, true);
+    }
+
     @Override
     public void updateServiceParameters(String serviceName, Map<String, Object> parameters) {
-        CloudServiceInstance service = getServiceInstance(serviceName);
+        CloudServiceInstance service = getServiceInstanceWithoutAuxiliaryContent(serviceName);
+        var updateServiceInstanceRequest = UpdateServiceInstanceRequest.builder()
+                                                                       .serviceInstanceId(service.getGuid()
+                                                                                                 .toString());
         if (service.isUserProvided()) {
-            updateUserProvidedServiceParameters(service, parameters);
-            return;
+            updateServiceInstanceRequest.credentials(parameters);
+        } else {
+            updateServiceInstanceRequest.parameters(parameters);
         }
-        delegate.serviceInstances()
-                .update(org.cloudfoundry.client.v2.serviceinstances.UpdateServiceInstanceRequest.builder()
-                                                                                                .serviceInstanceId(service.getGuid()
-                                                                                                                          .toString())
-                                                                                                .parameters(parameters)
-                                                                                                .acceptsIncomplete(true)
-                                                                                                .build())
-                .block();
-    }
-
-    private void updateUserProvidedServiceParameters(CloudServiceInstance service, Map<String, Object> parameters) {
-        delegate.userProvidedServiceInstances()
-                .update(UpdateUserProvidedServiceInstanceRequest.builder()
-                                                                .userProvidedServiceInstanceId(service.getGuid()
-                                                                                                      .toString())
-                                                                .credentials(parameters)
-                                                                .build())
+        delegate.serviceInstancesV3()
+                .update(updateServiceInstanceRequest.build())
```

Comment:
If it is a managed service instance, updating the parameters will return an async response https://v3-apidocs.cloudfoundry.org/version/3.119.0/#update-a-service-instance 
The job ID should be handled conditionally

Reply:
Again last operation is used for the polling